plugin/db/mysql/mysql.go

```diff
@@ -938,6 +948,68 @@ func (driver *Driver) Restore(ctx context.Context, sc *bufio.Scanner) (err error
 	return nil
 }
 
+func genRestoreFuncWithGhostTable(fnExecSQL func(query string) error) func(string) error {
+	reInsert := regexp.MustCompile(`INSERT INTO ` +
+		// Capture group for optional db prefix like "`dbPrefix`."
+		// the ?: means DO NOT capture this group
+		// the last ? means the whole db prefix is optional
+		"(?:`([^`]+)`\\.)?" +
+		// Capture group for the table name, leaving back quote ` outside for convenience
+		"`([^`]+)` " +
+		// Capture group for values to the end of the line
+		"VALUES (.*)")
+
+	reCreate := regexp.MustCompile("CREATE TABLE " +
+		// Capture group for the table name, leaving back quote ` outside for convenience
+		"`([^`]+)` " +
+		// Capture group for the rest of the contents
+		// Note that there will be multiple lines, so using [\s\S]+ to match them all including the line break
+		"([\\s\\S]+)")
+
+	// only matches name after CONSTRAINT qualifier
+	reConstraintName := regexp.MustCompile("CONSTRAINT `[^`]+`")
+	fn := func(stmt string) error {
+		// matchesInsert would be like:
+		// with dbPrefix: []string{"INSERT INTO `dbPrefix`.`table` VALUES (1, 2, 3);", "dbPrefix", "table", "(1, 2, 3);"}
+		// empty dbPrefix: []string{"INSERT INTO `table` VALUES (1, 2, 3);", "", "table", "(1, 2, 3);"}
+		matchesInsert := reInsert.FindStringSubmatch(stmt)
+
+		// matchesCreate would be like:
+		// []string{"CREATE TABLE `table` (", "table", "("}
+		matchesCreate := reCreate.FindStringSubmatch(stmt)
+
+		var stmtWithGhostTables string
+		if matchesInsert != nil {
+			dbPrefix := ""
+			if matchesInsert[1] != "" {
+				dbPrefix = fmt.Sprintf("`%s`.", matchesInsert[1])
+			}
+			ghostTable := fmt.Sprintf("%s_ghost", matchesInsert[2])
+			values := matchesInsert[3]
+			stmtWithGhostTables = fmt.Sprintf("INSERT INTO %s`%s` VALUES %s", dbPrefix, ghostTable, values)
+		} else if matchesCreate != nil {
+			// In the create table case, we must remove the constraint name from the SQL
+			// (foreign key/check constraints), or a duplicate constraint error will be raised by MySQL.
+			ghostTable := fmt.Sprintf("%s_ghost", matchesCreate[1])
+			tableDef := matchesCreate[2]
+			// If there are constraint clauses, we should remove the constraint name to avoid conflict.
+			// According to the MySQL docs (https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html),
+			// "If the CONSTRAINT symbol clause is not defined, or a symbol is not included following the CONSTRAINT keyword,
+			// a constraint name name is generated automatically."
```

Comment:
```suggestion
			// a constraint name is generated automatically."
```

Reply:
What if we design a format and append timestamp? This allows us to identify the constraint created by us.

Reply:
Good idea. It is feasible since [the max length is 64](https://dev.mysql.com/doc/refman/8.0/en/identifier-length.html).

Reply:
The rough idea is to prepend `bb_` to the original constraint name and append a timestamp (to seconds). When the whole length exceeds 64, we truncate the original constraint name part.

Reply:
I may go with the _ghost_timestamp. The reason is we may enforce "chk_" prefix as the naming convention for our schema review guide and we don't want to break the rule ourselves.  https://bytebase.com/database-review-guide

Reply:
After an offline discussion, me and @tianzhou agreed to append *_$timestamp* at the end. Before that, we should match against existing timestamp postfix and replace it. This behavior can mostly preserve the original constraint name, and enforce the same naming convension.